src/backend/distributed/operations/shard_transfer.c

```diff
@@ -1937,3 +1963,43 @@ WorkerApplyShardDDLCommandList(List *ddlCommandList, int64 shardId)
 
 	return applyDDLCommandList;
 }
+
+
+void
+UpdatePlacementUpdateStatusForShardIntervalList(List *shardIntervalList,
+												char *sourceName, int sourcePort,
+												PlacementUpdateStatus status)
+{
+	ProgressMonitorData *header = GetCurrentProgressMonitor();
```

Comment:
I don't think this will work for moves initiated by the rebalance monitor. Because in that case the progress monitor is created by that backend and not by this one. You should be able to get all progress monitors using ProgressMonitorList (for example usage see the implementation of `get_rebalance_progress`).